src/utils/echo-http-server.js

```diff
@@ -43,11 +43,21 @@ module.exports.createServer = () => {
 
   const server = http.createServer(handler)
 
-  server.start = (opts) => new Promise(
-    (resolve) => server.listen(Object.assign({ port: defaultPort, host: '127.0.0.1' }, opts), resolve)
-  )
+  server.start = (opts) => new Promise((resolve, reject) => {
+    server.once('error', reject)
+    server.listen(Object.assign({ port: defaultPort, host: '127.0.0.1' }, opts), () => {
+      server.removeListener('error', reject)
+      resolve()
+    })
+  })
 
-  server.stop = () => new Promise((resolve) => server.close(resolve))
+  server.stop = () => new Promise((resolve, reject) => {
+    server.once('error', reject)
+    server.close(() => {
```

Comment:
the callback can be called with an Error if the server was not open 

https://github.com/nodejs/node/blob/9cfb3841785cd658bb13369ed642247e089a5533/lib/net.js#L1519

Reply:
I saw that on docs but I thought `server.once('error', reject)` would call its callback on any error (including the `[ERR_SERVER_NOT_RUNNING]`).

Reply:
Btw, do we really need `error` event in this case?